### PR TITLE
mds: misc small patches

### PR DESCRIPTION
**Pull request by jcsp**
Various tiny things that spun off from #10390, hopefully all un-controversial.  Anything here that is controversial I can split into another PR.

### CODE REVIEW

File: src/mds/MDCache.cc [hunk 1]
```diff
@@ -448,13 +448,6 @@ void MDCache::create_mydir_hierarchy(MDSGather *gather)
     stray->store_backtrace(gather->new_sub());
   }
 
-  CInode *journal = create_system_inode(MDS_INO_LOG_OFFSET + mds->whoami, S_IFREG);
-  string name = "journal";
-  CDentry *jdn = mydir->add_primary_dentry(name, journal);
-  jdn->_mark_dirty(mds->mdlog->get_current_segment());
-
-  mydir->fnode.fragstat.nfiles++;
-  mydir->fnode.rstat.rfiles++;
```
Comment (gregsfortytwo) [inline src/mds/MDCache.cc]: This bit makes me more nervous than the earlier patches. I think we hardcode some assumptions about the value of these nfiles/rfiles counters in a few places; did you check for any issues with that?

**Comment by gregsfortytwo** [comment]
Rest of these look good.

File: src/mds/MDCache.cc (hunk 1 above)
Comment (jcsp) [inline src/mds/MDCache.cc]: No issues in the fs suite from this.  Can you be more specific about assumptions elsewhere?  Hopefully nothing is assuming that the nfiles/rfiles are anything other than the number of dentries?

Comment (gregsfortytwo) [inline src/mds/MDCache.cc]: I thought we had a few places (in vestigial shutdown code or whatever)
where we were trying to verify that we had cleaned stuff up by checking
that the number of recursive files was less than 3 (or whatever). I can't
find any now so I'm probably mixing it up with other things. :)

**Pull request merged**